app/controllers/admin/email_controller.rb

```diff
@@ -69,6 +69,19 @@ def send_digest
     end
   end
 
+  def smtp_should_reject
+    params.require(:from)
+    params.require(:to)
+    # These strings aren't localized; they are sent to an anonymous SMTP user.
+    if !User.exists?(email: Email.downcase(params[:from])) && !SiteSetting.enable_staged_users
+        render json: { reject: true, reason: "Mail from your address is not accepted. Do you have an account here?" }
```

Comment:
Minor style nitpick here, we follow the convention of 2 whitespaces when indenting our code.